gplog/gplog.go

```diff
@@ -120,31 +122,42 @@ func GetLogger() *GpLogger {
 // stdout and stderr are passed in to this function to enable output redirection in tests.
 func NewLogger(stdout io.Writer, stderr io.Writer, logFile io.Writer, logFileName string, verbosity int, program string) *GpLogger {
 	return &GpLogger{
-		logStdout:   log.New(stdout, "", 0),
-		logStderr:   log.New(stderr, "", 0),
-		logFile:     log.New(logFile, "", 0),
-		logFileName: logFileName,
-		verbosity:   verbosity,
-		header:      getHeader(program),
+		logStdout:     log.New(stdout, "", 0),
+		logStderr:     log.New(stderr, "", 0),
+		logFile:       log.New(logFile, "", 0),
+		logFileName:   logFileName,
+		verbosity:     verbosity,
+		header:        GetHeader(program),
+		logPrefixFunc: nil,
```

Comment:
Let's set this to `defaultLogPrefixFunc` instead of nil, so that `GetLogPrefix` can just return the current value without a check.

Reply:
Hmm.  Looking around, there's no good way to compare functions in Go by default.  This is probably fine, then, and we might want to add some function-comparison logic to the `structmatcher` package so we can use that in future tests.